crates/oxc_syntax/src/scope.rs

```diff
@@ -42,6 +42,10 @@ impl ScopeFlags {
         self.contains(Self::StrictMode)
     }
 
+    pub fn is_block(&self) -> bool {
```

Comment:
I don't think this is correct. `*self == Self::StrictMode` can be true when in a function but not within a block within that function.

Reply:
That's what I want, I just want to know if in the BlockStatement

Reply:
Yes, I think just `self.is_empty()` is what you need for that.

With `is_block` written as it is, `is_block()` would be true when inside `foo` here:

```
'use strict';
function foo() {
  // `ctx.scope().is_block()` is `true`.
  // `ctx.scope().is_empty()` is `false`.
  // `ctx.scope().is_function()` is `true`.
}
```

Reply:
Just to say: I don't like the design of `ScopeFlags`. I think it's very confusing!

Reply:
> Yes, I think just `ScopeFlags::is_empty` is what you need for that.
> 
> `ctx.scope().is_empty()` answers the question "am I in a nested block here?"
> 
> With `is_block` written as it is, `is_block()` would be true when inside `foo` here:
> 
> ```
> 'use strict';
> function foo() {
>   // `ctx.scope().is_block()` is `true`.
>   // `ctx.scope().is_empty()` is `false`.
>   // `ctx.scope().is_function()` is `true`.
> }
> ```

No. BlockStatement will inherit StrictMode. So actually result is

```ts
'use strict';
function foo() {
  // ScopeFlags::StrictMode | ScopeFlags::Function
  // `ctx.scope().is_block()` is `false`.
  // `ctx.scope().is_empty()` is `false`.
  // `ctx.scope().is_function()` is `true`.

  {
     // BlockStatement
     // ScopeFlags::StrictMode
     // `ctx.scope().is_block()` is `true`.
     // `ctx.scope().is_empty()` is `false`.
  }
}
```

You can check in [Playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAIDsgYCAgICAgICTHcrm5BJ12ajhv4gxb1aOKaBCbo9ZL1gx%2BdsX%2B2Uf9RZ%2FjolWJDOoDIMVPusJRxBm6byPrmkDH4xfhqNZbKCJEC7QmR2R2tH8zOQ%2Bu9NBF2tjJyUsHbE3dbo%2FTXe%2BmNWJbHOoIGeyk5W3jC9o34UYRQlHUnyW7zoifwuClYagYuI2LSMmVd90oCcdKBuG1rQeCwPz5ayL0uVjEk6%2Bf1%2FYerI%3D)

Reply:
Ah ha! Now I understand why we disagree. We're talking at cross-purposes. You are familiar with `ScopeFlags` from semantic, whereas I'm not.

I copied how `ScopeFlags` inherit from parent scope in `Traverse` from `VisitMut`. But I think `VisitMut`'s logic for it is different from semantic's.

So I am talking about how `ScopeFlags` behaves in `Traverse`, whereas you are talking about how it behaves in `Semantic`. We are both right, but talking about different things!

Obviously, this misalignment is bad. We should change the behavior in both `VisitMut` and `Traverse` to match `Semantic`, and then this confusion will go away.

Reply:
Oh no. Actually scratch all that. I was talking rubbish. `self.is_empty() || *self == Self::StrictMode` is correct, I just read it wrong.

There *is* a misalignment between `Semantic` and `Traverse`, but that's not the problem here.

Sorry I've wasted your time with this.

Reply:
Haha, It's okay. By the way, I don't like the implementation of `is_block`. I think we should add `ScopeFlags::Block` instead of `ScopeFlags::Empty`.  That will be easy to implement `is_block`